### PR TITLE
feat(cache-redis): add support to nestjs v10

### PR DESCRIPTION
**Pull request by overbit**
Fixes: amplication/amplication#7240

## PR Details

- adds support for DSG v2+ and Nestjs v10+
- moved to `ioredis` base library instead of `redis` as base library to integrate with Redis

## PR Checklist

- [x] Tests for the changes have been added
- [x] `npm build` doesn't throw any error
- [x] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: plugins/cache-redis/README.md [hunk 1]
```diff
@@ -32,12 +33,11 @@ If no configuration is provided the .amplicationrc.json file will use be used as
 
 ```json
 {
-    "host": "localhost",
-    "port": 6379,
-    "ttl": 5,
-    "max": 100,
```
Comment (mulygottlieb) [inline plugins/cache-redis/README.md:38]: Why was "max" removed if it's still part of the .amplicationrc?

File: plugins/cache-redis/README.md [hunk 2]
```diff
@@ -57,16 +57,15 @@ Running `npm run test` will run the plugin's test suite.
...
+
 To configure, set the following environment variables:
 
 REDIS_HOST - The host on which the Redis server is running.
 
 REDIS_PORT - The port on which the Redis server is listening.
 
-REDIS_TTL - The Time To Live for cached requests on the Redis server.
-
-REDIS_MAX_REQUESTS_CACHED - The largest amount of requests that can be cached at once.
```
Comment (mulygottlieb) [inline plugins/cache-redis/README.md:69]: same here regarding `max`

File: plugins/cache-redis/README.md (hunk 1 above)
Comment (overbit) [reply plugins/cache-redis/README.md:38]: good spot, removing it as it's not needed by the new library

File: plugins/cache-redis/README.md (hunk 2 above)
Comment (overbit) [reply plugins/cache-redis/README.md:69]: same

**Review by mulygottlieb — APPROVED** [review]

**Pull request merged**